compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/calls/CallableReferenceResolution.kt

```diff
@@ -396,10 +395,67 @@ fun ConeKotlinType.isKCallableType(): Boolean {
 private fun createKPropertyType(
     propertyOrField: FirVariable<*>,
     receiverType: ConeKotlinType?,
-    returnTypeRef: FirResolvedTypeRef
+    returnTypeRef: FirResolvedTypeRef,
+    callInfo: CallInfo,
 ): ConeKotlinType {
     val propertyType = returnTypeRef.type
     return org.jetbrains.kotlin.fir.resolve.createKPropertyType(
-        receiverType, propertyType, isMutable = propertyOrField.isVar
+        receiverType,
+        propertyType,
+        isMutable = propertyOrField.canBeMutableReference(callInfo)
     )
 }
+
+private fun FirVariable<*>.canBeMutableReference(callInfo: CallInfo): Boolean {
+    if (!isVar) return false
+    if (this is FirField) return true
+    if (this is FirProperty && this.hasJvmFieldAnnotation) return true
```

Comment:
Probably, this check is redundant
JvmField properties cannot have custom setters

Reply:
Right, not sure what regressions made me believe this is needed. Removing this doesn't affect anything.